appsmithorg/appsmith
# fix: fixed URL validation to not restrict navigateTo URLs to limited protocols

## Description
Currently, we check to see if the URL that is passed to the `navigateTo` method starts with http(s), mailTo or tel. This limits the capability to load/navigate to other URLs that do not conform to these protocols. This PR remove this crude protocol check and only checks to see if the navigateTo argument is a valid URL.
>
> Links to Notion, Figma or any other documents that might be relevant to the PR
>
>
#### PR fixes following issue(s)
Fixes #4878 
>
>
#### Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video
>
>
#### Type of change
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
>
>
>
## Testing
>
#### How Has This Been Tested?
>
>
#### Test Plan
1. Verify NavigateTo() for https:// mailTo ftp urls
2. Verify NavigateTo for above protocol from jsobject and from Action selector
3. Verify NavigateTo for same and NewWindow with above URLs

#### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)
>
>
>
## Checklist:
#### Dev activity


#### QA activity:
